### ISSUE TITLE
Configure Gradle build scan plugin

### ISSUE THREAD

**Issue by chetanmeh** [apache/incubator-openwhisk#3312]
Gradle provide a [hosted service][1] where build results can be published. This requires [build scan][2] plugin to be configured. 

- https://docs.gradle.com/build-scan-plugin
- https://gradle.com/build-scans
- Sample report from JUnit Project - https://scans.gradle.com/s/pgjgssca2kkli/tests
- Introduction Blog https://blog.gradle.org/introducing-build-scans

This would be specially useful when CI fails for some random test issues and also get an insight on which test is taking more time 

[1]: https://scans.gradle.com/get-started
[2]: https://plugins.gradle.org/plugin/com.gradle.build-scan/1.11

**Comment by chetanmeh** [comment]
As per of setup one needs to [accept terms of service][1] via 

```
buildScan {
    termsOfServiceUrl = 'https://gradle.com/legal/terms-of-service'
    termsOfServiceAgree = 'yes'
    publishAlwaysIf(System.getenv('CI') != null)
}
```
So has some legal implication. Checking the usage in Apache project it appears that [Groovy project has this configured][2] so it should be ok to use

[1]: https://docs.gradle.com/build-scan-plugin/#accept_the_scans_gradle_com_terms_of_service
[2]: https://github.com/apache/groovy/blob/270709706102e4efef5a59da0f185f48436a8402/build.gradle#L54

**Comment by chetanmeh** [comment]
Here is an output based on build from the fork https://gradle.com/s/cw3dfrmz237ig

**Comment by markusthoemmes** [comment]
Sweeeet, didn't know this existed. Can we hook this up with travis as indicated by 

> This would be specially useful when CI fails for some random test issues and also get an insight on which test is taking more time

**Comment by chetanmeh** [comment]
To find the scan report url look for log entry in travis like below

```
Publishing build scan...
https://gradle.com/s/reldo4qqlg3ka
```

**Comment by chetanmeh** [comment]
Closed via #3314
